tiledb/sm/subarray/range_subset.h

```diff
@@ -141,7 +140,78 @@ struct SortStrategy<std::string, std::string> {
           return a.start_str() < b.start_str() ||
                  (a.start_str() == b.start_str() && a.end_str() < b.end_str());
         });
-    return Status::Ok();
+  };
+};
+
+/** Default merge behavior: merging is not enabled. */
+template <typename T, typename Enable = T>
+struct MergeStrategy {
+  static void merge(std::vector<Range>&);
```

Comment:
Rename to merge_sorted_ranges so people don't get confused that the ranges need to be sorted first.